Refactor Navigation

osahtout opened the pull request:
This pull request aims to update some UI to more closely match the mockups.

1. Updated the homepage subheader to be always visible during scrolling and seen as part of the top navbar
1. Updated the time box UI in the event card to have the number bolded.
1. Updated the time in the event info header to be consistent and have the day in front of the month

Also, @TheGreatMarkus refactored the navigation bars to make more sense. Before, the 'Navbar" would be at the bottom on mobile and at the top on desktop. The "header" would be at the top on mobile and not exist on desktop. Now, there is a top navbar, and a bottom navbar, and they change their appearance based on what the mockup shows


![NEWNEWScreenshot 2020-12-31 104726](https://user-images.githubusercontent.com/44760251/103416757-03ebfc80-4b56-11eb-883b-01ee24464a19.png)

![NEWNEWScreenshot 2020-12-31 104809](https://user-images.githubusercontent.com/44760251/103416767-09e1dd80-4b56-11eb-8580-4e6f1b37fc13.png)

![NEWNEWScreenshot 2020-12-31 104835](https://user-images.githubusercontent.com/44760251/103416773-0ea69180-4b56-11eb-9fcb-6fd733fa718d.png)

![Screen Shot 2020-12-19 at 11 03 50 PM](https://user-images.githubusercontent.com/44760251/102704869-05acea80-424f-11eb-8cb8-cfc23063b9d9.png)

As for @JimLau49's discovery, we do need to create another ticket because it affects more than just the sub nav:
![NEWNEWScreenshot 2020-12-31 104907](https://user-images.githubusercontent.com/44760251/103416811-3eee3000-4b56-11eb-9e42-b4d5439ae2cb.png)

osahtout committed:
refactor (mull-ui): update subheader in homepage

Updated the sub header in the homepage to not be scrollable and have a background

no refs

osahtout committed:
refactor (mull-ui): time box in event card

Update the time box in event card to look closer to mockups. Day is now bolder. It should consider 24h time clock

no refs

osahtout committed:
refactor (mull-ui): update event page time to match event card

Day is now infront of month to stay consistant with event card and the world

no refs

osahtout committed:
test (mull-ui): update tests for new changes

no refs

osahtout committed:
test (mull-ui): update snapshot

no refs

TheGreatMarkus commented:
Wow I've never need a branch name so long lmao

cristian-aldea commented on apps/mull-ui/src/app/components/event-card/event-card.tsx:
```diff
@@ -32,8 +32,10 @@ export const EventCard = ({ event, style = {} }: EventCardProps) => {
+        <div style={{ fontWeight: 'bolder' }}>{day} </div>
+        <div>{month}</div>
         <div>{time}</div>
+        <div>{merīdiem}</div>
```
Must the meridiem be on a separate line? I don't think it looks nice. And if it's not placed on a separate line, then it would simplify the logic you added in `apps/mull-ui/src/utilities.ts`

osahtout replied:
meridiem is no longer used

cristian-aldea commented on apps/mull-ui/src/app/components/event-card/event-card.scss:
```diff
@@ -24,6 +24,7 @@
   border-radius: 0.5rem;
   padding: 0.2rem;
   text-align: center;
+  width: 2.8rem;
```
It would be better to add padding instead of hardcoding the width if you want some spacing imo.

osahtout replied:
that has been addressed

cristian-aldea commented on apps/mull-ui/src/utilities.ts:
```diff
@@ -5,14 +5,22 @@
   const timeString = Intl.DateTimeFormat('en-us', {
     hour: '2-digit',
     minute: '2-digit',
-  }).format(date);
-
-  return { year: date.getFullYear(), month, day: date.getDate(), time: timeString };
+  })
+    .format(date)
+    .split(' ');
+  timeString.push(''); //to take the 24h time into account
```
One thing that is kinda dangerous with this is that we can't know what `Intl.DateTimeFormat`, it depends on the format accepted in the region the user lives in. This might have edge cases we aren't aware of

osahtout replied:
This has been addressed

cristian-aldea commented on apps/mull-ui/src/app/components/subnavigation-bar/subnavigation-bar.scss:
```diff
@@ -1,8 +1,23 @@
 
-.subnavigation-container {
-  margin-top: 1rem;
+.subnavigation-header-container {
+  background-color: $background-color;
+  box-shadow: 0 0.25rem 0.25rem rgba(0, 0, 0, 0.1);
+  height: 2.5rem;
+  left: 0;
+  margin-top: -1rem;
+  position: fixed;
+  width: 100%;
+  z-index: 1;
+}
+
+.subnavigation-text-container {
+  position: relative;
+  margin-top: 0.5rem;
+  margin: 0 auto;
+  align-items: center;
+  width: 100%;
   display: flex;
   justify-content: space-evenly;
 }
```
This styling is really hacky, what you've done is that you're hiding the box shadow of the top navigation bar with the -1rem top margin. I think it would be easier if we make it part of the top navbar and make it appear only when on the discover page.

This would be a bit complicated however since the top nav bar is the navbar on desktop and the header on mobile. Hence I think we might as well refactor those too so they make more sense. One should always be at the top, and the other should be at the bottom, only appearing on mobile.

What do you think @r-saba

r-saba replied:
I agree @TheGreatMarkus don't think it would be complicated to do either

r-saba replied:
I know we discussed this and I was told that the navigation should look like the one in desktop on both desktop and mobile and that the mockups would be updated to match. Did this change @osahtout ? Not that I mind but I would have done what @TheGreatMarkus mentioned then.

osahtout replied:
I think there's a huge misunderstanding from me and I must apologize. I honestly thought that it was meant that the sub nav would be a separate component as code instead of it being separate visually from the top-nav (unless there was another discussion that I can't remember). 

I think having the sub nav flushed with the scrollable page would cause the user to have difficulty distinguishing the nav and the event card. 

![image](https://user-images.githubusercontent.com/44760251/102914422-3d35b580-444e-11eb-9981-4d9c1306c962.png)

cristian-aldea replied:
The way I did it in my test branch isn't ideal either for the reason you mentioned, and it wasn't the approach I was initially going for. Let's discuss the way to tackle this problem during the daily

cristian-aldea replied:
To document, my approach is now to have the sub nav bar separate from the top navbar. However, the box shadow for the top navbar is disabled and instead place on the sub nav bar, giving the illusion that they are together.

cristian-aldea replied:
To clarify, I've changed back to something similar to what omar did, my initial approach ended up brining too much complexity to the styling and possibly make it easy to break in the future if the page layout needs to change.

cristian-aldea reviewed (changes requested):
Left a few suggestions to simplify stuff, I wanna have a talk with you about some potential refactor to simplify the way the navbars work, since its a bit wonky right now. The wonkiness isn't related to your changes though, but we might as well refactor that too

TheGreatMarkus commented:
As discussed, I'll be contributing my suggestions to this branch, and if Omar isn't okay with then we can revert the commits

cristian-aldea committed:
refactor(discover-page): Overhaul navigation

Replaced all existing navigation component with new ones whose scope is better defined:
- top-nav-bar: This component represents the top navigation bar, which is different on mobile/desktop
- bot-nav-bar: This component represents the bottom navigation bar, which only shows up on mobile
- sub-nav-bar: This component represent the tabs on the discover page

TheGreatMarkus commented:
My contributions can be found on this branch: `refactor/cris-test` if anyone is interested. 

I also took the liberty of refactoring the oddities we have with our nav bars

cristian-aldea committed:
fix(discover-page): Refine discover page styling

cindyslittleplanet commented:
Just a comment about a small issue found on PR #129 that can be fixed here.
I find that the Tabs (the words) are too small compared to menu buttons and the event cards for the desktop view.
They are currently at size 16, which is the same size as the title of each field to complete on the Create Event page.
So, it feels weird to have the words of the Tabs to be the same size as field titles. 

![image](https://user-images.githubusercontent.com/38992971/102947313-08971d80-4491-11eb-896a-cf7e01108fd7.png)

TheGreatMarkus commented:
> Just a comment about a small issue found on PR #129 that can be fixed here.

@cindyslittleplanet yeah this is a valid concern. Originally the text size for the event cards seemed too small for me to read comfortably, so I decided to bump it up a lil bit. Now the tabs should also be made a bit bigger to compensate.

osahtout committed:
refactor (mull-ui): apply Cristian and PO changes

The time shown is now updated based on the POs feedback and applied the refactoring Cristian made for navigation.

no refs

osahtout committed:
fix (mull-ui): fix ui linting issues

no refs

osahtout committed:
test (mull-ui): update navigation test

The changes from Cristians branch should now have the updated tests

no ref

osahtout committed:
test (mull-ui): upate snapshots

JimLau49 commented on apps/mull-ui/src/app/components/event-card/event-card.tsx:
```diff
@@ -32,8 +32,8 @@ export const EventCard = ({ event, style = {} }: EventCardProps) => {
       <div className="event-card-datetime" data-testid="event-card-datetime">
-        <div>{`${month} ${day}`}</div>
-        <div>{time}</div>
+        <div style={{ fontWeight: 'bolder' }}>{`${day} ${month.toUpperCase()}`}</div>
```
inline styling

osahtout replied:
removed

JimLau49 commented on apps/mull-ui/src/app/components/event-card/event-card.tsx:
```diff
@@ -32,8 +32,8 @@ export const EventCard = ({ event, style = {} }: EventCardProps) => {
-        <div>{`${month} ${day}`}</div>
-        <div>{time}</div>
+        <div style={{ fontWeight: 'bolder' }}>{`${day} ${month.toUpperCase()}`}</div>
+        <div style={{ fontWeight: 'normal' }}>{time.replace(/\s/g, '')}</div>
```
same as above

osahtout replied:
removed

JimLau49 reviewed (changes requested):
Desktop looking good for me but i have some comments about mobile view. 

On various mobile displays im getting the sub-nav bar clumped together as well as 'my events' is appearing line broken instead of on the same line

![image](https://user-images.githubusercontent.com/35787279/103366315-4b0fba00-4a90-11eb-83a0-b70f95b960ef.png)

osahtout committed:
fix (mull-ui): addressed comments

removed inline styling and fixed the sub nav on mobile where the words were to close to each other

no refs

osahtout commented:
> Desktop looking good for me but i have some comments about mobile view.
> 
> On various mobile displays im getting the sub-nav bar clumped together as well as 'my events' is appearing line broken instead of on the same line
> 
> ![image](https://user-images.githubusercontent.com/35787279/103366315-4b0fba00-4a90-11eb-83a0-b70f95b960ef.png)


Tried to fix it the best I can but at some point, it will still get too cramped and enter a new line but that's before the bottom nav bleeds out the screen

<img width="325" alt="Screen Shot 2020-12-30 at 11 56 24 AM" src="https://user-images.githubusercontent.com/44760251/103369932-6c28d880-4a99-11eb-8a6d-5d5f89926fe0.png">

TheGreatMarkus commented:
> Tried to fix it the best ...

Yeah the text might need to be made a bit smaller to fit on the screen in that case. Maybe another breakpoint is needed.

osahtout commented:
> > Tried to fix it the best ...
> 
> Yeah the text might need to be made a bit smaller to fit on the screen in that case. Maybe another breakpoint is needed.

Then that would mean that the navigation button would also need to change with that breakpoint since they're cut of from the screen

TheGreatMarkus commented:
> Then that would mean that the navigation button would also need to change with that breakpoint since they're cut of from the screen
I agree with this @JimLau49. This issue already exists on develop with the bottom nav bar. It should be addressed in another pr through a new breakpoint imo.

cristian-aldea committed:
refactor(navigation bars): Made sub navbar position fixed

Instead of my previous approach where sub navbar is part of the page and the entire page is scrollable except for the subnavbar, the subnavbar is no longer part of the page flow. This ended up simplifying a lot of the styling in app.tsx

cristian-aldea committed:
refactor(styling): Remove unnecessary calc

Since the variables being added ahve the same unit, calc is not needed, and scss operators can be used instead.

cristian-aldea committed:
refactor(styles): Rename old css and test names

JimLau49 reviewed (dismissed):
Thanks for fixing the suggested changes :)

RGPosadas reviewed (changes requested):
Nice work, the UI looks more refined and matches our mockups/our PO's vision better!
I also really appreciate the navigation refactoring, it is much easier to follow now than before 🚀 

I'm blocking due to US-1.8's failing E2E test (for all lines like [this](https://github.com/RGPosadas/Mull/blob/develop/apps/mull-ui-e2e/src/integration/US-1.8.spec.ts#L27)) since we now show DAY first before MONTH.

Otherwise, everything looks great 😄 
![image](https://user-images.githubusercontent.com/33991121/103470169-d01efb80-4d3c-11eb-987e-fddfd03195f3.png)

cristian-aldea committed:
fix(e2e): Fix US-1.8 e2e tests

TheGreatMarkus commented:
> Nice work, the UI looks more refined and matches our mockups/our PO's vision better!
> ...

Good catch. I fixed the e2e test and took the time to rebase the branch. Have another look @RGPosadas  @JimLau49

TheGreatMarkus commented:
e2e tests are fixed, linting and units works as before, have another look when you can @RGPosadas  @JimLau49
